// @blueprintjs/core/lib/esnext/components/tag/compoundTag.js
/*
 * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import classNames from "classnames";
import * as React from "react";
import { Classes, DISPLAYNAME_PREFIX, Utils } from "../../common";
import { isReactNodeEmpty } from "../../common/utils";
import { Icon } from "../icon/icon";
import { Text } from "../text/text";
import { TagRemoveButton } from "./tagRemoveButton";
/**
 * Compound tag component.
 *
 * @see https://blueprintjs.com/docs/#core/components/compound-tag
 */
export const CompoundTag = React.forwardRef((props, ref) => {
    const { active, children, className, fill, icon, intent, interactive, leftContent, large, minimal, onRemove, rightIcon, round, tabIndex = 0, ...htmlProps } = props;
    const isRemovable = Utils.isFunction(onRemove);
    const tagClasses = classNames(Classes.TAG, Classes.COMPOUND_TAG, Classes.intentClass(intent), {
        [Classes.ACTIVE]: active,
        [Classes.FILL]: fill,
        [Classes.INTERACTIVE]: interactive,
        [Classes.LARGE]: large,
        [Classes.MINIMAL]: minimal,
        [Classes.ROUND]: round,
    }, className);
    return (React.createElement("span", { ...htmlProps, className: tagClasses, tabIndex: interactive ? tabIndex : undefined, ref: ref },
        React.createElement("span", { className: Classes.COMPOUND_TAG_LEFT },
            React.createElement(Icon, { icon: icon }),
            React.createElement(Text, { className: classNames(Classes.COMPOUND_TAG_LEFT_CONTENT, Classes.FILL), tagName: "span" }, leftContent)),
        React.createElement("span", { className: Classes.COMPOUND_TAG_RIGHT },
            !isReactNodeEmpty(children) && (React.createElement(Text, { className: classNames(Classes.COMPOUND_TAG_RIGHT_CONTENT, Classes.FILL), tagName: "span" }, children)),
            React.createElement(Icon, { icon: rightIcon }),
            isRemovable && React.createElement(TagRemoveButton, { ...props }))));
});
CompoundTag.defaultProps = {
    active: false,
    fill: false,
    interactive: false,
    large: false,
    minimal: false,
    round: false,
};
CompoundTag.displayName = `${DISPLAYNAME_PREFIX}.CompoundTag`;
